### PR TITLE
feat(eslint-plugin): [space-infix-ops] missing error report for conditional types

### LINKED ISSUE #5033: Bug: [space-infix-ops] Missing error report for conditional types
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.6.4&sourceType=module&code=C4TwDgpgBAKhDOwA8MB8UC8soQB7AgDsATeKAIwHtKAbCAQ0IH5gAnAVwgC4Azem+BABQQA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AjAIZJxaEBmaAHlgPYwIgBcIcKKzKIAL5A&tsConfig=N4XyA

### Repro Code

```typescript
type Test<T> = T extends boolean?true:false
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/space-infix-ops": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

```ts
type Test<T> = T extends boolean ? true : false
```

### Actual Result

```ts
type Test<T> = T extends boolean?true:false
```

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.25.0` |
| `@typescript-eslint/parser`            | `5.25.0` |
| `TypeScript`                           | `4.6.4` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16.15.0` |

### PR DESCRIPTION
**Pull request by holazz**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5033
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

**Review by bradzacher — CHANGES_REQUESTED** [review]
We've found reusing selectors like this is very brittle and can easily break!

Instead of hacking the AST to reuse their internal code - let's just fork their logic - it'll be less lines and it'll mean we are better able to modify it in future if we need.

**Comment by holazz** [comment]
> Thanks for this!
> 
> We've found reusing selectors like this is very brittle and can easily break!
> 
> Instead of hacking the AST to reuse their internal code - let's just fork their logic - it'll be less lines and it'll mean we are better able to modify it in future if we need.

Sorry, I saw some such usage in the source code, I thought it was intentional. I have changed it, please review it.

**Comment by holazz** [comment]
is there any status update on this? 🤔

File: packages/eslint-plugin/tests/rules/space-infix-ops.test.ts [hunk 1]
```diff
@@ -977,6 +977,26 @@ ruleTester.run('space-infix-ops', rule, {
         },
       ],
     },
+    {
+      code: `
+        type Test<T> = T extends boolean?true:false
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/tests/rules/space-infix-ops.test.ts:982]: A few edge cases to test for...

* What if there's only a space before, not after? And vice-versa?
* Placing newlines at various points in the ternary, with and without spaces

**Review by JoshuaKGoldberg — DISMISSED** [review]
Code looks good, just hoping for some more tests. 🙂

**Comment by holazz** [comment]
@JoshuaKGoldberg I added some more test cases, please review!

**Pull request merged**